File: JavaScript/7js 1/calcType.js
/*
|--------------------------------------------------------------------------
| Find the calculation type
|--------------------------------------------------------------------------
|
*/

function calcType(a, b, res) {
   if (a + b == res) {
      return 'addition'
   } else if (a - b == res) {
      return 'subtraction'
   } else if (a * b == res) {
      return 'multiplication'
   } else {
      return 'division'
   }
}


const calcType = (Q, S, R) => Q + S - R ? Q - S - R ? Q * S - R ? 'division' : 'multiplication' : 'subtraction' : 'addition'


const calcType = (a, b, res) =>
   a + b === res ? 'addition' :
   a - b === res ? 'subtraction' :
   a * b === res ? 'multiplication' :
   a / b === res ? 'division' :
   null;


const calcType = (a, b, res) => ({
   [a + b]: `addition`,
   [a - b]: `subtraction`,
   [a * b]: `multiplication`,
   [a / b]: `division`
})[res];


function calcType(a, b, res) {
   if (a < res && b < res) {
      return a + b == res ? 'addition' : 'multiplication'
   } else {
      return a - b == res ? 'subtraction' : 'division'
   }
}


console.log(calcType(1, 2, 3), 'addition');
console.log(calcType(10, 4, 40), 'multiplication');
console.log(calcType(10, 5, 5), 'subtraction');
console.log(calcType(9, 5, 1.8), 'division');

File: Python/7py 1/calc_type.py
"""
|--------------------------------------------------------------------------
| Find the calculation type
|--------------------------------------------------------------------------
|
"""

def calc_type(a, b, res):
   if (a + b == res) :
      return 'addition'
   elif (a - b == res) :
      return 'subtraction'
   elif (a * b == res) :
      return 'multiplication'
   else :
      return 'division'


def calc_type(a, b, res):
   return {a + b: "addition", a - b: "subtraction", a * b: "multiplication", a / b: "division"}[res]


def calc_type(a, b, res):
   dict = {a+b:'addition',a-b:'subtraction',a*b:'multiplication',a/b:'division'}
   return dict[res]


def calc_type(a, b, res):
   return next(x for op, x in [('+', 'addition'), ('-', 'subtraction'), ('*', 'multiplication'), ('/', 'division')] 
   if eval(f'{a}{op}{b}') == res)


print(calc_type(1, 2, 3), "addition")
print(calc_type(10, 5, 5), "subtraction")
print(calc_type(10, 4, 40), "multiplication")
print(calc_type(9, 5, 1.8), "division")